### ISSUE TITLE
onPin won't fire on page up.

### ISSUE THREAD

**Issue by kikoanis** [KyleAMathews/react-headroom#41]
So to reproduce this. You need to page down once and then page up. 
I did some debugging and I think this is the part that is responsible for the issue

```
if currentScrollY <= 0 and
      state.state isnt "unfixed"
    return {
      action: "unfix"
      scrollDirection: scrollDirection
      distanceScrolled: distanceScrolled
    }
```

currentScrollY in this case is 0 and the whole condition is true and so the action would be unfixed. Not sure if there should be another action for bin in this case. But if you modified the condition to 
```
if currentScrollY < 0 and
      state.state isnt "unfixed"
    return {
      action: "unfix"
      scrollDirection: scrollDirection
      distanceScrolled: distanceScrolled
    }
```
this won't satisfy the condition and it will fire the pin event

**Comment by KyleAMathews** [comment]
Thanks for the bug report! Do you think you'd like to take a stab at
creating a PR with a failing test case? And the fix too of course if you'd
like ;-)

**Comment by kikoanis** [comment]
Sure. Will try to submit a PR, may be by the end of today

**Comment by KyleAMathews** [comment]
Published 1.7.3
